examples/src/migration/migration_raw_rsa_key_example.py

```diff
@@ -0,0 +1,251 @@
+from aws_encryption_sdk.internal.crypto.wrapping_keys import WrappingKey
+from aws_encryption_sdk.key_providers.raw import RawMasterKeyProvider
+
+EXAMPLE_DATA: bytes = b"Hello World"
+
+DEFAULT_ENCRYPTION_CONTEXT : Dict[str, str] = {
+    "encryption": "context",
+    "is not": "secret",
+    "but adds": "useful metadata",
+    "that can help you": "be confident that",
+    "the data you are handling": "is what you think it is",
+}
+
+DEFAULT_KEY_NAME_SPACE = "Some managed raw keys"
+
+DEFAULT_KEY_NAME = "My 4096-bit RSA wrapping key"
+
+
+def generate_rsa_keys_helper():
+    """Generates a 4096-bit RSA public and private key pair
+
+    Usage: generate_rsa_keys_helper()
+    """
+    ssh_rsa_exponent = 65537
+    bit_strength = 4096
+    key = rsa.generate_private_key(
+        backend=crypto_default_backend(),
+        public_exponent=ssh_rsa_exponent,
+        key_size=bit_strength
+    )
+
+    # This example choses a particular type of encoding, format and encryption_algorithm
+    # Users can choose the PublicFormat, PrivateFormat and encryption_algorithm that align most
+    # with their use-cases
+    public_key = key.public_key().public_bytes(
+        encoding=crypto_serialization.Encoding.PEM,
+        format=crypto_serialization.PublicFormat.SubjectPublicKeyInfo
+    )
+    private_key = key.private_bytes(
+        encoding=crypto_serialization.Encoding.PEM,
+        format=crypto_serialization.PrivateFormat.TraditionalOpenSSL,
+        encryption_algorithm=crypto_serialization.NoEncryption()
+    )
+
+    return public_key, private_key
+
+
+DEFAULT_RSA_PUBLIC_KEY, DEFAULT_RSA_PRIVATE_KEY = generate_rsa_keys_helper()
+
+
+def create_keyring(public_key, private_key):
+    """Demonstrate how to create a Raw RSA keyring using the key pair.
+
+    Usage: create_keyring(public_key, private_key)
+    """
+    mat_prov: AwsCryptographicMaterialProviders = AwsCryptographicMaterialProviders(
+        config=MaterialProvidersConfig()
+    )
+
+    keyring_input: CreateRawRsaKeyringInput = CreateRawRsaKeyringInput(
+        key_namespace=DEFAULT_KEY_NAME_SPACE,
+        key_name=DEFAULT_KEY_NAME,
+        padding_scheme=PaddingScheme.OAEP_SHA256_MGF1,
+        public_key=public_key,
+        private_key=private_key
+    )
+
+    keyring: IKeyring = mat_prov.create_raw_rsa_keyring(
+        input=keyring_input
+    )
+
+    return keyring
+
+
+# This is a helper class necessary for the Raw RSA master key provider.
+# In the StaticRandomMasterKeyProvider, we fix the static key to
+# DEFAULT_RSA_PRIVATE_KEY in order to make the test deterministic.
+# Thus, both the Raw RSA keyring and Raw RSA MKP have the same private_key
+# and we are able to encrypt data using keyrings and decrypt using MKP and vice versa
+# In practice, users should generate a new key pair for each key id.
+class StaticRandomMasterKeyProvider(RawMasterKeyProvider):
+    """Randomly generates and provides 4096-bit RSA keys consistently per unique key id."""
```

Comment:
All the stuff about "Random" is a little inaccurate...
The key provider isn't randomly generating the keys, this module is.
The module defines a `DEFAULT_RSA_PRIVATE_KEY` variable, that this module uses...
Could you touch this up? I might just call it `StaticMasterKeyProvider`. The "random" aspect is abstracted from this class.